tap-agent/src/agent/sender_allocation.rs

```diff
@@ -508,6 +508,45 @@ impl SenderAllocationState {
                 ),
                 _ => e.into(),
             })?;
+
+        // store invalid receipts and if all are invalid remove them
+        if !invalid_receipts.is_empty() {
```

Comment:
you need to check if the valid receipts list is empty

Reply:
Hmmm I disagree, the logic before was okay in this, checking if there are `invalid_receipts` if so save them.
Now what I added inside of this if is that  if there are also no `valid_receipts` then yeah we remove them (the `invalid_receipts`).

Because if there are `valid_receipts` in there, then the rav creation process will take care of removing them

Reply:
Ok, I see that you added the `if valid_receipts.is_empty()` on line 525. But there's one problem that the older solution solved:

What if the request to tap-aggregator fails?

With the older solution, nothing would be stored on the database because it only stores it after the network request.